src/main/java/seedu/address/ui/InsuranceListPanel.java

```diff
@@ -77,13 +78,18 @@ private void handleSwitchToProfilePanelRequestEvent(SwitchToProfilePanelRequestE
         insuranceListView.getSelectionModel().clearSelection();
     }
 
-    /*@Subscribe
-    private void handleInsurancePanelSelectionChangedEvent(InsurancePanelSelectionChangedEvent event) {
-        InsuranceProfile selected = insuranceListView.getItems().filtered(insuranceProfile -> {
-            return insuranceProfile.getInsurance().equals(event.getInsurance());
-        }).get(0);
-        insuranceListView.getSelectionModel().select(selected);
-    }*/
+    @Subscribe
+    private void handleInsuranceClickedEvent(InsuranceClickedEvent event) {
+        ObservableList<InsuranceProfile> insurances = insuranceListView.getItems();
+        for (int i = 0; i < insurances.size(); i++) {
+            if (insurances.get(i).getInsurance().getInsuranceName().equals(event.getInsurance().getInsuranceName())) {
+                insuranceListView.getSelectionModel().select(i);
+                break;
+            }
+        }
+    }
+    // weird phenomenon that a filteredList does not contain elements in the original list and cannot be used
+    // in the select command
```

Comment:
Is that the reason the lambda expression is not used?

Reply:
yes exactly

Reply:
Understood.